components/raftstore/src/store/peer_storage.rs

```diff
@@ -1539,20 +1541,23 @@ mod tests {
 
     #[derive(Default)]
     struct ReadyContext {
-        kv_wb: WriteBatch,
+        kv_wb: Option<WriteBatch>,
         raft_wb: WriteBatch,
         sync_log: bool,
     }
 
     impl HandleRaftReadyContext for ReadyContext {
-        fn kv_wb(&self) -> &WriteBatch {
-            &self.kv_wb
+        fn take_kv_wb(&mut self) -> WriteBatch {
+            self.kv_wb.take().unwrap_or_default()
         }
-        fn kv_wb_mut(&mut self) -> &mut WriteBatch {
-            &mut self.kv_wb
+        fn save_kv_wb(&mut self, b: WriteBatch) {
+            self.kv_wb = Some(b);
         }
-        fn raft_wb(&self) -> &WriteBatch {
-            &self.raft_wb
+        fn kv_wb_mut(&mut self) -> &mut WriteBatch {
+            if self.kv_wb.is_none() {
+                self.kv_wb = Some(WriteBatch::new());
+            }
+            self.kv_wb.as_mut().unwrap()
```

Comment:
Like I mentioned elsewhere, this pattern of lazily creating a new writebatch is problematic for two reasons: it is imprecise about the invariants of when this type needs to be `Some`, and the `engine_traits` traits don't give `WriteBatch` it's own constructor.

I would much rather have this field not be an `Option` and always be available.